### PR TITLE
engine: fix write empty modifies issue.

### PR DESCRIPTION
**Pull request by disksing**
@BusyJay @siddontang @ngaut @zhangjinpeng1987 

The issue is when use async api to write and the modifies vec is empty, the Callback does not get called.

### CODE REVIEW

File: tests/storage/test_raftkv.rs [hunk 1]
```diff
@@ -156,3 +157,7 @@ fn cf(ctx: &Context, engine: &Engine) {
     must_delete_cf(ctx, engine, "cf", b"key");
     assert_none_cf(ctx, engine, "cf", b"key");
 }
+
+fn empty_write(ctx: &Context, engine: &Engine) {
+    engine.write(&ctx, vec![]).unwrap();
+}
```
Comment (BusyJay) [inline tests/storage/test_raftkv.rs]: Leave an empty line.

**Comment by BusyJay** [comment]
rest LGTM

**Comment by BusyJay** [comment]
LGTM

File: src/storage/engine/raftkv.rs [hunk 2]
```diff
@@ -182,9 +182,6 @@ impl<C: PdClient> Engine for RaftKv<C> {
                    mut modifies: Vec<Modify>,
                    cb: Callback<()>)
                    -> engine::Result<()> {
-        if modifies.is_empty() {
-            return Ok(());
```
Comment (siddontang) [inline src/storage/engine/raftkv.rs]: why not calling cb here with an empty error?

Comment (disksing) [inline src/storage/engine/raftkv.rs]: `cb` is supposed to be called in another thread, calling `cb` here will break the behavior and may lead to unexpected result.

**Comment by siddontang** [comment]
LGTM

**Pull request merged**